pocs/camera/zwo.py

```diff
@@ -0,0 +1,446 @@
+            msg = "Image type '{} not supported by {}".format(new_image_type, self.model)
+            self.logger.error(msg)
+            raise ValueError(msg)
+        roi_format = self._ASIDriver.get_roi_format(self._camera_ID)
+        roi_format['image_type'] = new_image_type
+        Camera._ASIDriver.set_roi_format(self._camera_ID, **roi_format)
+
+    @property
+    def ccd_temp(self):
+        """ Current temperature of the camera's image sensor """
+        return self._control_getter('TEMPERATURE')[0]
+
+    @property
+    def ccd_set_point(self):
+        """ Current value of the target temperature for the camera's image sensor cooling control.
+
+        Can be set by assigning an astropy.units.Quantity
+        """
+        return self._control_getter('TARGET_TEMP')[0]
+
+    @ccd_set_point.setter
+    def ccd_set_point(self, set_point):
+        self._control_setter('TARGET_TEMP', set_point)
+
+    @property
+    def ccd_cooling_enabled(self):
+        """ Current status of the camera's image sensor cooling system (enabled/disabled) """
+        return self._control_getter('COOLER_ON')[0]
+
+    @ccd_cooling_enabled.setter
+    def ccd_cooling_enabled(self, on):
+        self._control_setter('COOLER_ON', on)
+
+    @property
+    def ccd_cooling_power(self):
+        """ Current power level of the camera's image sensor cooling system (as a percentage). """
+        return self._control_getter('COOLER_POWER_PERC')[0]
+
+    @property
+    def gain(self):
+        """ Current value of the camera's gain setting in internal units.
+
+        See `egain` for the corresponding eletrons / ADU value.
+        """
+        return self._control_getter('GAIN')[0]
+
+    @gain.setter
+    def gain(self, gain):
+        self._control_setter('GAIN', gain)
+
+    @property
+    def egain(self):
+        """ Nominal value of the image sensor gain for the camera's current gain setting """
+        self._refresh_info()
+        return self._info['e_per_adu']
+
+    # Methods
+
+    def __str__(self):
+        # ZWO ASI cameras don't have a port so just include the serial number in the string
+        # representation.
+        s = "{} ({})".format(self.name, self.uid)
+
+        if self.focuser:
+            s += ' with {}'.format(self.focuser.name)
+            if self.filterwheel:
+                s += ' & {}'.format(self.filterwheel.name)
+        elif self.filterwheel:
+            s += ' with {}'.format(self.filterwheel.name)
+
+        return s
+
+    def connect(self):
+        """
+        Connect to ZWO ASI camera.
+
+        Gets 'camera_ID' (needed for all driver commands), camera properties and details
+        of available camera commands/parameters.
+        """
+        self.logger.debug("Connecting to {}".format(self))
+        self._refresh_info()
+        self._camera_ID = self._info['camera_ID']
+        self.model, _, _ = self._info['name'].partition('(')
+        if self._info['is_colour_camera']:
```

Comment:
I don't see `is_colour_camera` anywhere else. And should probably be `is_color_camera` for consistency (sorry! :smile: )

Reply:
*monocle drops out*